mmdet3d/models/necks/dla_neck.py

```diff
@@ -0,0 +1,208 @@
+            Default: None.
+    """
+
+    def __init__(self,
+                 start_level,
+                 channels,
+                 scales,
+                 in_channels=None,
+                 norm_cfg=None):
+        super(DLAUp, self).__init__()
+        self.start_level = start_level
+        if in_channels is None:
+            in_channels = channels
+        self.channels = channels
+        channels = list(channels)
+        scales = np.array(scales, dtype=int)
+        for i in range(len(channels) - 1):
+            j = -i - 2
+            setattr(
+                self, 'ida_{}'.format(i),
+                IDAUp(channels[j], in_channels[j:], scales[j:] // scales[j],
+                      norm_cfg))
+            scales[j + 1:] = scales[j]
+            in_channels[j + 1:] = [channels[j] for _ in channels[j + 1:]]
+
+    def forward(self, layers):
+        """Forward function.
+
+        Args:
+            layers (tuple[torch.Tensor]): Features from multi-scale layers.
+
+        Returns:
+            tuple[torch.Tensor]: Up-sampled features of different layers.
+        """
+        outs = [layers[-1]]
+        for i in range(len(layers) - self.start_level - 1):
+            ida = getattr(self, 'ida_{}'.format(i))
+            ida(layers, len(layers) - i - 2, len(layers))
+            outs.insert(0, layers[-1])
+        return outs
+
+
+@NECKS.register_module()
+class DLANeck(nn.Module):
+    """DLA Neck.
+
+    Args:
+        in_channels (list[int], optional): List of input channels
+            of multi-scale feature map.
+        start_level (int, optioanl): The scale level where upsampling
+            starts. Default: 2.
+        end_level (int, optional): The scale level where upsampling
+            ends. Default: 5.
+        norm_cfg (dict, optional): Config dict for normalization
+            layer. Default: None.
+    """
+
+    def __init__(self,
+                 in_channels=[16, 32, 64, 128, 256, 512],
+                 start_level=2,
+                 end_level=5,
+                 norm_cfg=None):
+        super(DLANeck, self).__init__()
+        self.start_level = start_level
+        self.end_level = end_level
+        scales = [2**i for i in range(len(in_channels[self.start_level:]))]
+        self.dla_up = DLAUp(
+            start_level=self.start_level,
+            channels=in_channels[self.start_level:],
+            scales=scales,
+            norm_cfg=norm_cfg)
+        self.ida_up = IDAUp(
+            in_channels[self.start_level],
+            in_channels[self.start_level:self.end_level],
+            [2**i for i in range(self.end_level - self.start_level)], norm_cfg)
+
+    def forward(self, x):
+        x = list(x)
+        x = self.dla_up(x)
+        outs = []
+        for i in range(self.end_level - self.start_level):
+            outs.append(x[i].clone())
+        self.ida_up(outs, 0, len(outs))
+        return outs[-1]
+
+    def init_weights(self):
```

Comment:
Should we directly inherit from `BaseModule` in this version? Or we need to further create a PR to refactor the initialization methods.

Reply:
init_weights function is necessary in the DLANeck module, the specially designed initial ways can benefit the training process and reduce the training epochs. refer to [https://github.com/open-mmlab/mmdetection/blob/master/mmdet/models/necks/ct_resnet_neck.py](url)